src/main/java/com/hedvig/notificationService/configuration/CustomerIOConfig.kt

```diff
@@ -4,28 +4,71 @@ import com.fasterxml.jackson.databind.ObjectMapper
 import com.hedvig.customerio.Customerio
 import com.hedvig.customerio.CustomerioClient
 import com.hedvig.customerio.CustomerioMock
+import com.hedvig.notificationService.customerio.ConfigurationProperties
+import com.hedvig.notificationService.customerio.MemberServiceImpl
+import com.hedvig.notificationService.customerio.ProductPricingFacade
+import com.hedvig.notificationService.customerio.ProductPricingFacadeImpl
+import com.hedvig.notificationService.customerio.Router
+import com.hedvig.notificationService.customerio.Workspace
+import com.hedvig.notificationService.serviceIntegration.memberService.MemberServiceClient
+import com.hedvig.notificationService.serviceIntegration.productPricing.client.ProductPricingClient
+import okhttp3.OkHttpClient
+import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Value
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 
 @Configuration
-class CustomerIOConfig {
+class CustomerIOConfig() {
 
-    @Bean("customerIO")
-    @ConditionalOnProperty("hedvig.usefakes", havingValue = "true", matchIfMissing = false)
-    fun customerIOMock(objectMapper: ObjectMapper): CustomerioClient {
-        return CustomerioMock(objectMapper)
+    @Autowired
+    lateinit var configuration: ConfigurationProperties
+
+    @Value("\${hedvig.usefakes:false}")
+    var useFakes: Boolean = false
+
+    private val okHttp: OkHttpClient = OkHttpClient()
+
+    @Bean("productPricingFacade")
```

Comment:
If the value is the qualifier I think it should be more explicit
```suggestion
    @Qualifier("productPricingFacade")
    @Bean()
```